src/collection-facets/more-facets-content.ts

```diff
@@ -0,0 +1,437 @@
+
+    this.collectionNameCache?.preloadIdentifiers(collectionIdsArray);
+  }
+
+  /**
+   * Converts the raw `aggregations` to `FacetGroups`, which are easier to use
+   */
+  private get aggregationFacetGroups(): FacetGroup[] {
+    const facetGroups: FacetGroup[] = [];
+    Object.entries(this.aggregations ?? []).forEach(([key, buckets]) => {
+      // the year_histogram data is in a different format so can't be handled here
+      if (key === 'year_histogram') return;
+
+      const option = getFacetOptionFromKey(key);
+      const title = facetTitles[option];
+      const castedBuckets = buckets.buckets as Bucket[];
+
+      // find length and pagination size for modal pagination
+      const { length } = Object.keys(castedBuckets as []);
+      this.paginationSize = Math.ceil(length / this.facetsPerPage);
+
+      // asynchronously load the collection name
+      if (option === 'collection') {
+        this.preloadCollectionNames(castedBuckets);
+      }
+
+      // render only items which will be visible as per this.facetsPerPage
+      const bucketsMaxSix = castedBuckets?.slice(
+        (this.pageNumber - 1) * this.facetsPerPage,
+        this.pageNumber * this.facetsPerPage
+      );
+
+      const facetBuckets: FacetBucket[] = bucketsMaxSix.map(bucket => {
+        let bucketKey = bucket.key;
+        // for languages, we need to search by language code instead of the
+        // display name, which is what we get from the search engine result
+        if (option === 'language') {
+          bucketKey =
+            this.languageCodeHandler?.getCodeStringFromLanguageName(
+              `${bucket.key}`
+            ) ?? bucket.key;
+        }
+        return {
+          displayText: `${bucket.key}`,
+          key: `${bucketKey}`,
+          count: bucket.doc_count,
+          state: 'none',
+        };
+      });
+      const group: FacetGroup = {
+        title,
+        key: option,
+        buckets: facetBuckets,
+      };
+      facetGroups.push(group);
+    });
+
+    return facetGroups;
+  }
+
+  private get getMoreFacetsTemplate(): TemplateResult {
+    return html`
+      <facets-template
+        .facetGroup=${this.mergedFacets?.shift()}
+        .selectedFacets=${this.selectedFacets}
+        .renderOn=${'modal'}
+        .collectionNameCache=${this.collectionNameCache}
+        @selectedFacetsChanged=${(e: CustomEvent) => {
+          this.selectedFacets = e.detail;
+        }}
+      ></facets-template>
+    `;
+  }
+
+  private get loaderTemplate(): TemplateResult {
+    return html`<div class="facets-loader">
+      <ia-activity-indicator .mode="processing"></ia-activity-indicator>
+    </div> `;
+  }
+
+  // render pagination if more then 1 page
+  private get facetsPaginationTemplate() {
+    return this.paginationSize > 1
+      ? html`<more-facets-pagination
+          .size=${this.paginationSize}
+          @pageNumberClicked=${this.pageNumberClicked}
```

Comment:
just a few questions around more facets pagination that i will add here:

Q: how do you handle size constraints for the page count display?
Q: is there a minimum amount of buttons do you show?
Q: what is the min width of a page button? 25px or 30px?

Q: re modal. is there minimum size when it's multi-column?
Q: what size does "multi-column" view happen?
Q: will/can resize observer be of use here to help calculate amount of room available for buttons?
Q:

Reply:
let's discuss in today's 1:1